### PR TITLE
fmt: Preserve leading comment spacing

### LINKED ISSUE #400: format: gradually eating up my spaces in the comments
I wouldn't want the formatter to touch the contents of my comments)

image

### PR DESCRIPTION
**Pull request by alixander**
fixes #400

### CODE REVIEW

**Review by nhooyr — CHANGES_REQUESTED** [review]

File: d2format/format.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func (p *printer) comment(c *d2ast.Comment) {
 	lines := strings.Split(c.Value, "\n")
 	for i, line := range lines {
 		p.sb.WriteString("#")
-		if line != "" && !strings.HasPrefix(line, " ") {
```
[the remark this answers is not in the document]
Comment (pleshevskiy) [reply d2format/format.go:84]: I confirm that this PR fixes the issue.

As an input to this function, strings with comments without # and one space go into this function. By removing this check, we always add one space in its place.

[the remark this answers is not in the document]
Comment (nhooyr) [reply d2format/format.go:84]: Ah right that makes sense, as otherwise when we parse again the leading space isn't parsed as part of the comment.

**Review by nhooyr — APPROVED** [review]
Thanks @pleshevskiy

File: d2format/format.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alixander) [reply d2format/format.go:84]: > This is supposed to just add a leading space to comments without leading spaces.

i added a test case for this: https://github.com/terrastruct/d2/pull/402/files#diff-f5d6f3315c026ce320a9cb3a4e2b86eb851bf4e3ddb65a2607bf996f8ab04409R603

**Pull request merged**